src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -1689,6 +1689,23 @@ void MacroAssembler::store_sized_value(Address dst, Register src, size_t size_in
   }
 }
 
+void MacroAssembler::load_word_misaligned(Register dst, Address src, Register tmp, bool is_signed) {
```

Comment:
I am afraid that the function name is a bit confusing considering that the well-known global 'wordSize' is 8 on linux-riscv64. But we are actually loading 4 bytes here. How about renaming it as something like "load_int_unaligned"? This would align with existing function names like 'load_unsigned_byte' and 'load_unsigned_short'. Also, it's safer to add an assertion to make sure that 'dst' and 'tmp' are different registers.